common/djangoapps/student/views.py

```diff
@@ -2654,3 +2687,24 @@ def get_context_data(self, **kwargs):
         })
 
         return context
+
+@csrf_exempt
```

Comment:
why do we need this?

Reply:
This view is for the ajax call. Django by default expects csrf token in the request otherwise it ignores the request.